### ISSUE TITLE
Name problem?

### ISSUE THREAD

**Issue by floDenk** [bbc/audio-offset-finder#33]
Hi together! 
Thank you for your very interesting program!
Wanted to try it out and installed it both from github and from pypi, however I had issues using the command line tool. Having a look at some of the paths, it looks like the naming of the script is a little odd:

![image](https://user-images.githubusercontent.com/82228164/183623383-5664c7a2-ae04-44f5-8569-f85210596b29.png)

When executing "executable-name.exe" I get the following error, which might be related to the issue above:

![image](https://user-images.githubusercontent.com/82228164/183623518-004b44e4-f677-4144-a01c-ab1c9a8b2a44.png)

Kind regards!
Flo

**Comment by stephenjolly** [comment]
Hi Flo,

I'm not a Windows user, but I'll do my best to help.  I'm not sure where "executable-name.exe" has come from.  I believe [there are tools](https://docs.python.org/3/faq/programming.html#faq-create-standalone-binary) that can convert python scripts to exe files, but I don't believe that we've put anything in the repository that would cause them to be used automatically.  Feel free to correct me if I'm mistaken about this though.

Can you check your `python\python38\scripts\` directory and see if there's a file in there named "audio-offset-finder"?  If so, try running that.

HTH,

S

**Comment by floDenk** [comment]
Hi Stephen, thanks for your reply! I indeed had a look at the folder where
the executables are located and did not find one named audio-offset-finder
only one named executable-name.exe which got installed with the
audio-offset-finder package...

Kind regards!
Flo


Stephen Jolly ***@***.***> schrieb am Mo., 15. Aug. 2022,
16:19:

**Comment by utz32** [comment]
seems like an issue with the last line in setup.cfg and the fact this project does not have a main() function.

**Comment by stephenjolly** [comment]
I've created https://github.com/bbc/audio-offset-finder/tree/module-path-errors to try and address these issues.  At the moment, this just includes a fix for the executable name - my expectation is that the ImportError will still be there.  Testing welcome, though.

@utz32 - I believe that there is a main() function defined by line 120 in bin/audio-offset-finder:

> if __name__ == "__main__":
>     sys.exit(main(sys.argv[1:]))

Again, I could be mistaken though.

**Comment by dreirund** [comment]
I also get this error when I run the executable:

```
/usr/lib/python3.10/site-packages/numba/np/ufunc/ufuncbuilder.py:327: NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details.
  self.nb_func = jit(_target='npyufunc', cache=cache)(py_func)
Traceback (most recent call last):
  File "/usr/bin/audio-offset-finder", line 5, in <module>
    from audio_offset_finder.audio_offset_finder import main
ImportError: cannot import name 'main' from 'audio_offset_finder.audio_offset_finder' (/usr/lib/python3.10/site-packages/audio_offset_finder/audio_offset_finder.py)
```

(On Arch Linux)

**Comment by davidvankemenade** [comment]
Hi @stephenjolly, I'm trying to use this very useful programme as part of a workflow in a Windows environment and I came across the same issue.

Normally, when I install programmes using "pip install .", I get a Windows executable in local-packages\Python311\Scripts. In this case, I would expect this to be audio_offset_finder.exe, but as @floDenk mentioned, it's executable-name.exe in this case. When I run this executable, I get the "main" issue mentioned before. This still happens with the https://github.com/bbc/audio-offset-finder/tree/module-path-errors tree.

I went back many commits, to https://github.com/bbc/audio-offset-finder/tree/a6b5e201ff8b8bb0b45eec05e5db438fc0e4f21f. When I "pip install ." this one, I get a "audio-offset-finder" (without .exe") exectuable in the Scripts directory. When I run "python audio-offset-finder" from this directory, I can run the (old version of the) programme.

Does this give any clues as to why this programme seems to behave differently from others in this respect? If it's possible to somehow solve this issue I'd be very grateful - otherwise I can still make good use of audio-offset-finder using the "python audio-offset-finder" workaround.

Kind regards,
David
